portal/src/main/webapp/js/api/cbioportal-datamanager.js

```diff
@@ -352,15 +353,36 @@ window.initDatamanager = function (genetic_profile_ids, oql_query, cancer_study_
 				return this.genetic_profile_ids;
 			},
 			'getSampleIds': function() {
-				return this.sample_ids;
+				var samples_ = [];
+				_.each(this.getStudySampleMap(), function(val,keyl) {
```

Comment:
It would also probably be useful to make it possible to pass the study id as an argument and get the samples for that study